src/FakeItEasy/Creation/IMethodInterceptionTester.cs

```diff
@@ -0,0 +1,16 @@
+namespace FakeItEasy.Creation
+{
+    using System.Reflection;
+
+    internal interface IMethodInterceptionTester
+    {
+        /// <summary>
+        /// Gets a value indicating whether the specified method can be intercanepted by  instance.
+        /// </summary>
+        /// <param name="method">The member to test.</param>
+        /// <param name="callTarget">The instance the method will be called on.</param>
+        /// <param name="failReason">The reason the method can not be intercepted.</param>
+        /// <returns>True if the member can be intercepted.</returns>
+        bool MethodCanBeInterceptedOnInstance(MethodInfo method, object callTarget, out string failReason);

```

Comment:
Looks like this interface has the same method as `CastleDynamicProxyInterceptionValidator`. Maybe `CastleDynamicProxyInterceptionValidator` could implement `IMethodInterceptionTester` (which could be renamed to `IMethodInterceptionValidator`), and be called directly without going through the `ProxyGenerator`?